packages/frontend/src/components/ExportCSV/index.tsx

```diff
@@ -45,41 +46,48 @@ export type ExportCSVProps = {
 
 const ExportCSV: FC<ExportCSVProps> = memo(
     ({ rows, getCsvLink, isDialogBody, renderDialogActions }) => {
-        const { showToastError } = useToaster();
+        const { showToastError, showToastInfo } = useToaster();
 
         const [limit, setLimit] = useState<string>(Limit.TABLE);
         const [customLimit, setCustomLimit] = useState<number>(1);
         const [format, setFormat] = useState<string>(Values.FORMATTED);
-        const [isExporting, setIsExporting] = useState<boolean>(false);
+
+        const { isFetching: isExporting, refetch: handleExport } = useQuery(
+            [limit, customLimit, rows, format],
+            () => {
+                showToastInfo({
+                    title: 'Fetching results. This might take a while...',
+                });
```

Comment:
just tested locally and I have few suggestions on how to improve the UI/UX of the tooltip.

1. intent of the toast should be primary (blue colour)
2. we should show a loading spinner instead of the icon. you can use blueprint's spinner for that (so users understand it is a loading state)
3. you can remove the default timeout so the tooltip stays open while it's processing.
4. you can pass a key to this method and then dismiss on success
5. you can call `AppToast.dismiss(KEY)` on `onSettled`.

Reply:
let me know if I you have questions or need help 🙌